plugins/inputs/arista_cloudvision_telemtry/arista_cloudvision_telemetry.go

```diff
@@ -0,0 +1,644 @@
+type CVP struct {
+	// CVP information
+	Cvpaddress    string         `toml:"addresses"`
+	Subscriptions []Subscription `toml:"subscription"`
+
+	Encoding    string
+	Origin      string //origin openconfig is supported today only.
+	Prefix      string
+	UpdatesOnly bool `toml:"updates_only"`
+
+	Cvptoken string `toml:"cvptoken"` // JSON web toke for ClouVision
+
+	// Redial
+	Redial config.Duration
+
+	// GRPC TLS settings
+	EnableTLS bool `toml:"enable_tls"`
+	internaltls.ClientConfig
+
+	// Internal state
+	internalAliases map[string]string
+	acc             telegraf.Accumulator
+	cancel          context.CancelFunc
+	wg              sync.WaitGroup
+	// Lookup/device+name/key/value
+	lookup      map[string]map[string]map[string]interface{}
+	lookupMutex sync.Mutex
+
+	// Device target -> name
+	devices map[string]string
+
+	Log telegraf.Logger
+}
+
+type Subscription struct {
+	Name   string
+	Origin string //origin openconfig is supported today only.
+	Path   string
+
+	// Subscription mode and interval
+	SubscriptionMode string          `toml:"subscription_mode"`
+	SampleInterval   config.Duration `toml:"sample_interval"`
+
+	// Duplicate suppression
+	SuppressRedundant bool            `toml:"suppress_redundant"`
+	HeartbeatInterval config.Duration `toml:"heartbeat_interval"`
+}
+
+// Struct for cloudvision API to return device data.
+type CvPDevices struct {
+	Result struct {
+		Value struct {
+			Key struct {
+				DeviceID string `json:"deviceId"`
+			} `json:"key"`
+			SoftwareVersion    string    `json:"softwareVersion"`
+			ModelName          string    `json:"modelName"`
+			HardwareRevision   string    `json:"hardwareRevision"`
+			Fqdn               string    `json:"fqdn"`
+			Hostname           string    `json:"hostname"`
+			DomainName         string    `json:"domainName"`
+			SystemMacAddress   string    `json:"systemMacAddress"`
+			BootTime           time.Time `json:"bootTime"`
+			StreamingStatus    string    `json:"streamingStatus"`
+			ExtendedAttributes struct {
+				FeatureEnabled struct {
+					Danz bool `json:"Danz"`
+					Mlag bool `json:"Mlag"`
+				} `json:"featureEnabled"`
+			} `json:"extendedAttributes"`
+		} `json:"value"`
+		Time time.Time `json:"time"`
+		Type string    `json:"type"`
+	} `json:"result"`
+}
+
+func (*CVP) SampleConfig() string {
+	return sampleConfig
+}
+
+// Start the CVP gNMI telemetry service
+func (c *CVP) Start(acc telegraf.Accumulator) error {
+	var err error
+	var ctx context.Context
+	var tlscfg *tls.Config
+	c.acc = acc
+	ctx, c.cancel = context.WithCancel(context.Background())
+	c.lookupMutex.Lock()
+	c.lookup = make(map[string]map[string]map[string]interface{})
+	c.devices = make(map[string]string)
+	c.lookupMutex.Unlock()
+
+	// Parse TLS config
+	if c.EnableTLS {
+		if tlscfg, err = c.ClientConfig.TLSConfig(); err != nil {
+			return err
+		}
+	}
+
+	if c.Cvptoken != "" {
+		tlscfg = &tls.Config{
+			Renegotiation:      tls.RenegotiateNever,
+			InsecureSkipVerify: true,
```

Comment:
This is for the outgoing gNMI connections.  It is unfortunate but right now we only support insecure on this gNMI connection.  Which will change.

Reply:
Honestly speaking, I would prefer to have this as an option and uncomment it in the sample.conf like
```toml
[[inputs.arista_cloudvision_telemtry]]
  ## CVP Address
  addresses = "10.255.35.170:8443"
  ## redial in case of failures after
  redial = "10s"

  enable_tls = false
  insecure_skip_verify = true

  ## cvp service account access token generated at /cv/settings/aaa-service-accounts
  cvptoken = "-snip-"

  [[inputs.arista_cloudvision_telemtry.subscription]]
    ## Name of the measurement
    name = "System"
    origin = "openconfig"
    path = "/interfaces/interface/state/counters"
    subscription_mode = "target_defined" 
```
to make it explicit and obvious to the user. Otherwise, people might assume certificates are checked...